impl/src/main/java/org/jboss/weld/bootstrap/enablement/EnablementListView.java

```diff
@@ -105,23 +106,26 @@ public boolean removeAll(Collection<?> c) {
         if (getExtension() != null) {
             BootstrapLogger.LOG.typeModifiedInAfterTypeDiscovery(getExtension(), c, REMOVE_OPERATION, getViewType());
         }
-        return getDelegate().removeAll(c);
+        // use impl from AbstractCollection, that one will still invoke our contains() method
+        return super.removeAll(c);
     }
 
     @Override
     public boolean remove(Object o) {
         if (getExtension() != null) {
             BootstrapLogger.LOG.typeModifiedInAfterTypeDiscovery(getExtension(), o, REMOVE_OPERATION, getViewType());
         }
-        return getDelegate().remove(o);
+        // implementation of Item does equals/hashCode only based on Java class
+        return getDelegate().remove(createSource((Class<?>) o, 0));
```

Comment:
Why don't you use `objectToItemIfNeeded()` here?

Reply:
Eeehm, because I implemented this method before introducing the helper one.
Will fix this, thanks!